### DISCUSSION TITLE
req.query object randomly sorted

### DISCUSSION THREAD

**Discussion by Malthehave** [vercel/next.js#22873] [Help]
Hello,

I have an api route set up at `/api/install`

Inside of that API route I access the users query parameters, with `req.query` . So when someone accesses: `/api/install?shop=myshop.com&timestamp=1615210067`, the expected output of  `req.query` would be : 
```
{
  shop: 'myshop.com',
  timestamp: '1615210067'
}
```
This is the case when developing locally on my own machine, but when I push this code to my production environment, which is Vercel, the output of  `req.query` is: 
```
{
  timestamp: '1615210067',
  shop: 'myshop.com'
}
```
So it's in a different order, despite the exact same query in the url: `/api/install?shop=myshop.com&timestamp=1615210067` . This is a problem since I need to validate an hmac based on the order the query parameters were input.

Do you know if this is the expected behavior, and thus you can't rely on the order of the `req.query` object? Or maybe this is something specifically with Vercels serverless functions?

In any case I would love to hear your thoughts, and ideas on how I might solve this.
Thanks in advance!

**Comment by Asjas** [comment]
`req.query` is an object and Objects are an unsorted data structure in JavaScript. You cannot determine the order of the keys. 🙂

**Reply by Malthehave** [reply]
Thanks for your reply! That makes sense, I suppose I have to sort the dictionary based on the keys myself. 
But isn't it weird that the order is different based on the environment the code is run in?
